stubs/redis/redis/client.pyi

```diff
@@ -599,28 +601,121 @@ class Redis(Generic[_StrType]):
     def zadd(
         self, name: _Key, mapping: Mapping[_Key, _Value], nx: bool = ..., xx: bool = ..., ch: bool = ..., incr: bool = ...
     ) -> int: ...
-    def zcard(self, name): ...
+    def zcard(self, name: _Key) -> int: ...
     def zcount(self, name: _Key, min: _Value, max: _Value) -> int: ...
-    def zincrby(self, name, value, amount=...): ...
-    def zinterstore(self, dest, keys, aggregate=...): ...
-    def zlexcount(self, name, min, max): ...
-    def zpopmax(self, name, count=...): ...
-    def zpopmin(self, name, count=...): ...
-    def bzpopmax(self, keys, timeout=...): ...
-    def bzpopmin(self, keys, timeout=...): ...
-    def zrange(self, name, start, end, desc=..., withscores=..., score_cast_func=...): ...
-    def zrangebylex(self, name, min, max, start=..., num=...): ...
-    def zrangebyscore(self, name, min, max, start=..., num=..., withscores=..., score_cast_func=...): ...
-    def zrank(self, name: _Key, value: _Key) -> Optional[int]: ...
-    def zrem(self, name, *values): ...
-    def zremrangebylex(self, name, min, max): ...
-    def zremrangebyrank(self, name, min, max): ...
+    def zincrby(self, name: _Key, value: _Value, amount: float = ...) -> float: ...
+    def zinterstore(self, dest: _Key, keys: Iterable[_Key], aggregate: Literal["SUM", "MIN", "MAX"] = ...) -> int: ...
+    def zlexcount(self, name: _Key, min: _Value, max: _Value): ...
+    def zpopmax(self, name: _Key, count: int = ...) -> List[_StrType]: ...
+    def zpopmin(self, name: _Key, count: int = ...) -> List[_StrType]: ...
+    @overload
+    def bzpopmax(self, keys: Union[_Key, Iterable[_Key]], timeout: Literal[0] = ...) -> Tuple[_StrType, _StrType, float]: ...
+    @overload
+    def bzpopmax(self, keys: Union[_Key, Iterable[_Key]], timeout: float) -> Optional[Tuple[_StrType, _StrType, float]]: ...
+    @overload
+    def bzpopmin(self, keys: Union[_Key, Iterable[_Key]], timeout: Literal[0] = ...) -> Tuple[_StrType, _StrType, float]: ...
+    @overload
+    def bzpopmin(self, keys: Union[_Key, Iterable[_Key]], timeout: float) -> Optional[Tuple[_StrType, _StrType, float]]: ...
+    @overload
+    def zrange(
+        self,
+        name: _Key,
+        start: int,
+        end: int,
+        desc: bool = ...,
+        *,
+        withscores: Literal[True],
+        score_cast_func: Callable[[float], _ScoreCastFuncReturn] = ...,
+    ) -> List[Tuple[_StrType, _ScoreCastFuncReturn]]: ...
+    @overload
+    def zrange(
+        self,
+        name: _Key,
+        start: int,
+        end: int,
+        desc: bool = ...,
+        withscores: bool = ...,
+        score_cast_func: Callable[[Any], Any] = ...,
+    ) -> List[_StrType]: ...
```

Comment:
I wasn't sure about how to type this method.

When `withscores` is true, the return type is a tuple. I removed the `= ...` to support this behavior, but that required me making withscores kwargs.